packages/engine/src/engine-errors.ts

```
@@ -0,0 +1,269 @@
+
+  // Timeout errors
+  if (/ETIMEDOUT|timeout.*connection|connection.*timeout|deadline exceeded|timed out after \d+ms/i.test(message)) {
+    return new TimeoutError(message, undefined, undefined, err instanceof Error ? err : undefined);
+  }
+
+  // 5xx / service unavailable
+  if (/upstream connect error|disconnect\/reset before headers|remote connection failure|transport failure/i.test(message)) {
+    return new ServiceUnavailableError(message, undefined, undefined, err instanceof Error ? err : undefined);
+  }
+
+  if (/"type":"server_error"|\"code\":\"server_error\"/i.test(message)) {
```

ESLint

[error] Unnecessary escape character: \".

(no-useless-escape)